### DISCUSSION TITLE
Why in KafkaBuilder are const assinged Ports

### DISCUSSION THREAD

**Discussion by Exoil** [testcontainers/testcontainers-dotnet#915] [Q&A] [answered]
Hello,
I have question about const values for kafka ports. Because When I use dedicated container builder "KafkaBuilder ", I would like change these values. There is some possibility to change it by testcontainers?
```

 public const ushort KafkaPort = 9092;

    public const ushort BrokerPort = 9093;

    public const ushort ZookeeperPort = 2181;
```

**Comment by HofmeisterAn** [comment, marked as answer]
These are the default ports of the Kafka service (container ports). The container ports are mapped to public random host ports. This is recommended and considered from us as best practices described [here](https://dotnet.testcontainers.org/api/create_docker_network/#exposing-container-ports-to-the-host) and [here](https://dotnet.testcontainers.org/api/best_practices/).

Modules usually do not allow configuring the ports of the service running inside the container (container ports). This applies for the Kafka module too, and we do not think this is even necessary. From our experiences and discussions we had in the past, there is not a real use case. Can you elaborate your use case? OC, you can always use the generic builder to configure the containers as you like.
